test/rule_test_support.dart

```diff
@@ -472,7 +471,12 @@ abstract class _ContextResolutionTest with ResourceProviderMixin {
   List<String> get collectionIncludedPaths;
 
   /// The analysis errors that were computed during analysis.
-  List<AnalysisError> get errors => result.errors;
+  List<AnalysisError> get errors => result.errors
+      .whereNot((e) => ignoredErrorCodes.any((c) => e.errorCode == c))
+      .toList();
+
+  /// Error codes that by default should be ignored in test expectations.
+  List<HintCode> get ignoredErrorCodes => [HintCode.UNUSED_LOCAL_VARIABLE];
```

Comment:
Please use `WarningCode.UNUSED_LOCAL_VARIABLE` as this HintCode will (🤞 🤞 🤞 ) be removed soon. https://github.com/dart-lang/sdk/blob/main/pkg/analyzer/lib/src/error/codes.g.dart#L7112

Reply:
Oh, yeah!  Totally.